crates/store/src/state.rs

```diff
@@ -574,3 +571,15 @@ async fn load_accounts(
     SimpleSmt::with_leaves(account_data)
         .map_err(StateInitializationError::FailedToCreateAccountsTree)
 }
+
+fn compute_tx_hash(updated_accounts: &[BlockAccountUpdate]) -> Digest {
```

Comment:
This logic for this function is very similar to the one for `BlockWitness`, should it be factored out to avoid problems if it changes? (although likely tests would catch it so if it's more convenient this way it's fine)

Reply:
The logic is similar, but it operates with different structures with different layouts. Thus, it's not easy to write abstractions to deal with both of them. Luckily, one of the methods is being used to check another one, so if one of them is changed, this check will fail.

Reply:
I do agree with @igamigo that having this logic in two different places is not ideal. And now that I think about it, we should probably move it to `miden-base`. Specifically, we could do the following:

1. Implement a generic function `compute_tx_hash()` which takes an iterator over `(TransactionId, AccountId)` tuples and returns the hash.
2. Implement `Block::transactions()` method and `BlockWitness::transactions()` methods which would return iterators over these tuples.
3. I would probably also implement `Block::compute_tx_hash()` method which returns the tx hash.